plugins/connection/docker.py

```diff
@@ -220,16 +216,19 @@ def _build_exec_cmd(self, cmd):
 
         return local_cmd
 
-    def _set_conn_data(self):
-
-        ''' initialize for the connection, cannot do only in init since all data is not ready at that point '''
-
+    def _set_docker_args(self):
         # TODO: this is mostly for backwards compatibility, play_context is used as fallback for older versions
         # docker arguments
         del self._docker_args[:]
-        extra_args = self.get_option('docker_extra_args') or self._play_context.docker_extra_args
+        extra_args = self.get_option('docker_extra_args') or getattr(self._play_context, 'docker_extra_args', '')
         if extra_args:
-            self._docker_args += extra_args.split(' ')
+            self._docker_args = shlex.split(extra_args)
```

Comment:
This is a breaking change. If we merge this, then only in a separate PR (that cannot be backported).

Reply:
more of a bugfix, since now it will actually handle quotes correctly

Reply:
This is still a breaking change, the behavior was different for many years now, folks could have relied on this. Fixing this in a 2.x bugfix release is ok for me, but this shouldn't be changed in the stable-1 branch. And it should definitely not be mixed with totally unrelated bugfixes.

Reply:
I removed this in 228b5bc00362e55c199ef7977eb244f7b4190a3c.